pkg/cloudprovider/fake/factory.go

```diff
@@ -28,13 +27,16 @@ var (
 
 type Factory struct {
 	WantErr error
-	// NodeReplicas is use by tests to control observed replicas.
-	NodeReplicas map[string]int32
+	// NodeReplicas is used by tests to control observed replicas.
+	NodeReplicas     map[string]*int32
+	NodeGroupStable  bool
+	NodeGroupMessage string
```

Comment:
We might be able to get away without testing this. It's a cosmetic feature; the behavior is driven by the boolean